### PR TITLE
[WIP ]Adding thank you page in order process

### PR DESCRIPTION
**Pull request by antonioperic**
I talked with @pjedrzejewski about that we need to have really order complete/thankyou page and not redirect to homepage with flash message.

I started to adding this step, do you think this is good way, or maybe we should pull in some other way?

In order page will be message about status and some order information

Thanks!

### CODE REVIEW

**Comment by antonioperic** [comment]
@pjedrzejewski what is status with your update?

File: src/Sylius/Bundle/WebBundle/Resources/views/Frontend/Checkout/Step/thankYou.html.twig [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {% if order.state == 'confirmed' %}
+        <h1>{{ 'sylius.checkout.thank_you.header'|trans|raw }}</h1>
+    {% else %}
+        <h1>{{ 'sylius.checkout.thank_you.header'|trans|raw }}</h1>
+    {% endif %}
+</div>
+
+<div class="row">
+    <div class="col-md-12">
+        <h4>Thank you for your order</h4>
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/views/Frontend/Checkout/Step/thankYou.html.twig]: I guess those texts should also be translated...

Comment (antonioperic) [inline src/Sylius/Bundle/WebBundle/Resources/views/Frontend/Checkout/Step/thankYou.html.twig]: I agree, this was WIP, but I stopped regarding @pjedrzejewski comments above. If he already has this completed then I think we should wait

**Comment by antonioperic** [comment]
@pjedrzejewski what is status with this? This is like very important feature.

**Comment by colejarczyk** [comment]
@pjedrzejewski are you thinking about affiliate programs in this separate purchase controller? There should be a possibility to inject partners JS code. ThankYouPage is perfect place to monitor our sales.

**Comment by antonioperic** [comment]
Affiliate will be amazing thing. For current project that we are working on we are also building affiliate and loyalty program. But still waiting for @pjedrzejewski to see what is the status with this thankyou page feature.

**Comment by antonioperic** [comment]
@pjedrzejewski do you have any update about this issue?

**Comment by stloyd** [comment]
@pjedrzejewski ping ;)

**Comment by antonioperic** [comment]
@pjedrzejewski anything on this?

File: src/Sylius/Bundle/CoreBundle/Checkout/Step/ThankYouStep.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        $context->close();
+
+        $event = new OrderCompleteEvent($order);
+        $this->dispatchEvent(SyliusCheckoutEvents::PURCHASE_COMPLETE, $event);
+
+        if ($event->hasResponse()) {
+            return $event->getResponse();
+        }
+
```
Comment (cu-syamsudin) [inline src/Sylius/Bundle/CoreBundle/Checkout/Step/ThankYouStep.php]: There should be SyliusCheckoutEvents::THANK_YOU_COMPLETE event dsipatched here right? Else the cart won't get abandoned

Comment (sudent) [inline src/Sylius/Bundle/CoreBundle/Checkout/Step/ThankYouStep.php]: There should be SyliusCheckoutEvents::THANK_YOU_COMPLETE event dsipatched here right? Else the cart won't get abandoned

**Comment by sudent** [comment]
I'm using this currently and it's working for my use case. However, would be great if it gets updated with the latest code.

File: src/Sylius/Bundle/CoreBundle/EventListener/OrderThankYouListener.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        if (!in_array(
+            $event->getSubject()->getState(),
+            array(
+                PaymentInterface::STATE_PENDING,
+                PaymentInterface::STATE_PROCESSING,
+                PaymentInterface::STATE_COMPLETED
+            )
+        )
+        ) {
```
Comment (sudent) [inline src/Sylius/Bundle/CoreBundle/EventListener/OrderThankYouListener.php]: In array or not in array..?

**Comment by antonioperic** [comment]
@sudent  this principle can do job, but I stopped to work on this as @pjedrzejewski said that he is already preparing something more complex on this theme. Unfortunately still there is no update

**Comment by nicolasricci** [comment]
@pjedrzejewski @michalmarcinkowski can we please have an update on this feature? This sounds just crazy that Sylius does not handle a thank you page at the end of the process.
I have implemented @antonioperic solution which works fine but if anything else is on its way it would be good to know.

File: src/Sylius/Bundle/CoreBundle/Checkout/Step/PurchaseStep.php [hunk 4]
```diff
@@ -81,12 +81,7 @@ public function forwardAction(ProcessContextInterface $context)
 
         $this->getDoctrine()->getManager()->flush();
 
-        $event = new PurchaseCompleteEvent($payment);
-        $this->dispatchEvent(SyliusCheckoutEvents::PURCHASE_COMPLETE, $event);
```
Comment (patrick-mcdougle) [inline src/Sylius/Bundle/CoreBundle/Checkout/Step/PurchaseStep.php]: Why can't you just listen on the purchase complete event and return the thank you page?

**Comment by aramalipoor** [comment]
@pjedrzejewski What is the status of this Thank You page feature? Is there anything I can help with?

**Comment by pjedrzejewski** [comment]
@aramalipoor Definitely! We basically need a separate controller action, to which you would be redirected after checkout. In process scenario, instead of redirecting to homepage, we would go to that page and there simply render a template with customer's ``order`` available. For now it can be super simple. Let me know if you will start something about that, then we can discuss more. Thank you!

**Comment by pjedrzejewski** [comment]
@aramalipoor The difference to this PR should be that it is not part of the process anymore, it is a separate page and the process should be completed already when you get redirected.

**Comment by aramalipoor** [comment]
@pjedrzejewski Great. I'll try to come up with something in coming days ;) The only thing is that how can I get order (or its id) to pass to `setRedirectParams` of ProcessBuilder [on this line](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php#L62)

**Comment by pjedrzejewski** [comment]
I think we can set the order id in the session. Or you can redirect with id, but then be sure to check if it belongs to the user to avoid exposing all order thank you pages. :D But I think session would be better.

**Comment by michalmarcinkowski** [comment]
Thanks Antonio! This will be included in https://github.com/Sylius/Sylius/issues/3710.

**Pull request closed without merging**